repos/keg-components/src/components/list/sectionList.native.js

```diff
@@ -0,0 +1,394 @@
+import { View } from 'KegView'
+import PropTypes from 'prop-types'
+import { Text } from '../typography/text'
+import { useScroll } from 'KegUseScroll'
+import { useClassName } from 'KegClassName'
+import { useThemePath } from '../../hooks/useThemePath'
+import { useScrollClassName } from 'KegScrollClassName'
+import { checkCall, noPropObj, noPropArr, get, isFunc } from '@keg-hub/jsutils'
+import React, {
+  useCallback,
+  useRef,
+  useMemo,
+  useEffect,
+  useState,
+  createRef,
+} from 'react'
+import { SectionList as RNSectionList, SafeAreaView } from 'react-native'
+import {
+  renderFromType,
+  getElementLayout,
+  scrollList,
+} from '../../utils/components'
+
+/**
+ * Helper hook to add __kegIndex to the sections
+ * This allows referencing them as needed without requiring the consumer to add the index
+ *
+ * @param {Array} sections - Groups of items to be displayed in the SectionList component
+ *
+ * @return {Array} - sections array with the __kegIndex added
+ */
+const useIndexedSections = (sections, indexBy) => {
+  return useMemo(() => {
+    return sections.map((section, index) => ({
+      ...section,
+      __kegIndex: get(section, indexBy, index),
+    }))
+  }, [sections])
+}
+
+/**
+ * Helper hook to allow tracking scrolling between section
+ * @param {boolean} doScrolling - Is automatic scrolling turned on
```

Comment:
- `doScrolling` && `listRef `is not part of the param anymore it seems.
- also missing `activeSection` && `setActiveSection` `@param`

Reply:
updated